Update to prost v11 and tonic v9

jkilpatr opened the pull request:
The big jump in prost v11 was that protoc was no longer bundled, meaning you now need an up to date version installed on your machine in order to compile the protos. Furthermore the well known google protos are no longer bundled, these protos come with a normal installation or protoc so I've included them in the repo for the case of simplicity.

Other changes:
* updated all crates to rust 2021
* renamed the 'staking' module to the 'distribution' module in cosmos-sdk for whatever reason the old staking name has been kept for the module only the filename has changed
* Added updated compile instructions to the readme to make the new requirements clear.

jkilpatr committed:
Add github actions + gitignore

jkilpatr committed:
Update bech32 ibc pin

jkilpatr committed:
Update to Prost v11 and Tonic v9 [1/3]

This patch contains the dependency updates and cargo.toml changes for
updating the prost and tonic versions to their respective latest.

The biggest change in this version is that Prost no longer bundles the
protoc compiler or the well known protos which leaves us to update our
tooling to handle them.

jkilpatr committed:
Add common_proto folder [2/3]

This patch adds the common proto folder for common google protos which
are not included by default in all of the repos who's protos we are
consuming because prost previously bunlded them and they are generally
bundled with protoc installations.

This patch includes those common protos in this repo for ease of
reproducibility.

jkilpatr committed:
Add newly generated files for Prost v11 and Tonic v9 [3/3]

ChristianBorst commented on README.md:
```diff
@@ -13,9 +21,9 @@ gravity_proto: https://crates.io/crates/gravity_proto
-bech32-ibc: https://github.com/althea-net/althea-chain  
-cosmos-sdk: https://github.com/althea-net/althea-chain  
-gravity: https://github.com/althea-net/althea-chain  
+bech32-ibc: https://github.com/althea-net/bech32-ibc  
```
Nice catch

ChristianBorst commented on cosmos_sdk_proto/src/prost/cosmos.staking.v1beta1.rs:
```diff
@@ -1,1281 +0,0 @@
-/// owned by one delegator, and is associated with the voting power of one
-/// validator.
-#[derive(Clone, PartialEq, ::prost::Message)]
-pub struct Delegation {
```
Staking and Distribution are not equivalent, for example this struct does not exist in  cosmos.distribution.v1beta1.rs

ChristianBorst committed:
Re-add problematic cosmos.staking.v1beta1.rs file

The source definition for CosmosSDK's staking protos has a namespace
conflict in authz.proto. Both a `message` and a `oneof` are given the
same name, which causes a namespace conflict with prost but not for the
Go compilation. We manually track this file as a band-aid solution for
now.

jkilpatr committed:
Fix staking/distribution module reference error

The staking module is not generated automatically because it has errors. In my confusion I thought 'distribution' had replaced 'staking' and pointed this reference at the wrong code.

ChristianBorst committed:
Always compile CosmosSDK staking module protos, add disclaimer

As mentioned in the README.md, the staking module can be particularly
difficult to work with since it always compiles to have a namespace
conflict. The README mentions that file should usually be ignored due to
disuse of the conflicting values, with an explanation of the manual work
to fix the conflict.
